Add friends likes

masmetus opened the pull request.

masmetus committed:
Вынесена логика из контроллеров в сервисы.
Создан уровень хранилища и интерфейс для работы с ним
Реализована работа с фильмами
Добавлен хендлер для ошибок + DTO

masmetus committed:
Фиксы для фильма + реализована логика друзей

z8q commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,141 @@
+            oldFilm.setName(newFilm.getName());
+        }
+
+        if (newFilm.getReleaseDate() != null) {
+            oldFilm.setReleaseDate(newFilm.getReleaseDate());
+        }
+
+        if (newFilm.getDescription() != null && !newFilm.getDescription().isBlank()) {
+            oldFilm.setDescription(newFilm.getDescription());
+        }
```
у нас в контроллере используется put аннотация для обновлений
бывает еще patch

от put ожидается, что объект обновится целиком, либо не обновится, если например поля не проходят валидацию

от patch ожидается, что мы можем заполнить часть полей, и объект обновится только по заполненным значениям, остальные останутся преждними

тут пример
https://www.baeldung.com/cs/http-put-vs-patch

к тому, что тут похоже можно принять объект целиком на обновление, либо не принять
можно еще с наставником посоветоваться

masmetus committed:
Фиксы после ревью

masmetus committed:
Фикс импортов

masmetus committed:
Обновление моделий под будущую реализацию БД и добавлены Enum классы для значений, что неизменны.
Добавлен README.md файл с логической моделью + описанием проекта и примером sql запросов

Andrey-K10 reviewed (commented):
Все хорошо, замечаний нет.
